tokenizers/src/tokenizer/mod.rs

```diff
@@ -507,20 +507,28 @@ impl Tokenizer {
         // 5. Convert offsets back to original string
         let mut current_offset = (0, 0);
         let mut n_source = &normalized;
-        output
-            .get_offsets_mut()
-            .iter_mut()
-            .for_each(|(start, end)| {
-                if (*start, *end) < current_offset {
-                    n_source = &pair_normalized.as_ref().unwrap_or(&normalized);
-                }
-                current_offset = (*start, *end);
-                let (s, e) = n_source
-                    .convert_offsets(Range::Normalized(*start..*end))
-                    .map_or((*start, *end), |range| (range.start, range.end));
-                *start = s;
-                *end = e;
-            });
+
+        let mut convert_offsets_back = |encoding: &mut Encoding| {
+            encoding
+                .get_offsets_mut()
+                .iter_mut()
+                .for_each(|(start, end)| {
+                    if (*start, *end) < current_offset {
```

Comment:
I think the initial `n_source` should be inside the closure (and `current_offset` too I guess), otherwise, it will use the `pair_normalized` after the first one that needed it.

Reply:
Oh yes you're right, I just forced-push again with the fix!